istioctl/pkg/tag/generate.go

```diff
@@ -257,20 +257,6 @@ base:
 	return whBuf.String(), nil
 }
 
-// mergeMaps merges maps into one. If both maps have the same key, the value from override will be used.
-func mergeMaps(base, override map[string]string) map[string]string {
-	if base == nil {
-		return override
-	}
```

Comment:
I think the original intention here is to save resource consumption, but this case will cause users to be troubled by whether the base has been modified, which I think is unnecessary

Reply:
I do think this should be handled if it becomes a more general function. It's not for memory allocation, but rather to avoid panicking due to wrong usage. Additionally, if this is a refactoring, it shouldn't change the original behavior.

Reply:
If we want to avoid panic caused by nil in the function, we must have a return value, because 
if the input parameter of map nil, we cannot persist modifications to parameter map outside the function